src/ds/thread_messaging.h

```diff
@@ -267,16 +267,21 @@ namespace threading
 
     void run()
     {
-      Task& task = tasks[get_current_thread_id()];
+      Task& task = get_task(get_current_thread_id());
 
       while (!is_finished())
       {
         task.run_next_task();
       }
     }
 
-    Task& get_task(uint16_t tid)
+    inline Task& get_task(uint16_t tid)
     {
+      CCF_ASSERT_FMT(
+        tid <= thread_count,
```

Comment:
Should this be `<` rather than `<=`? If we have 3 threads, the highest valid (0-indexed) `tid` is `2`?

Reply:
thank you. Fixed